RiotSwiftUI/Modules/Room/LocationSharing/MapViewErrorAlertInfoBuilder.swift

```diff
@@ -0,0 +1,47 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+struct MapViewErrorAlertInfoBuilder {
+    
+    func build(with error: LocationSharingViewError, primaryButtonCompletion: (() -> Void)?) -> AlertInfo<LocationSharingAlertType>? {
```

Comment:
Let's maybe rename this `primaryButtonCompletion` to `dimissalCallback` as the caller isn't aware of primary and secondary buttons.